File: app/src/main/java/com/snowdango/numac/AppListView/AppListViewActivity.java
package com.snowdango.numac.AppListView;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.Display;
import android.view.WindowManager;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.snowdango.numac.NumAcMain.NumAcActivity;
import com.snowdango.numac.R;
import com.snowdango.numac.NumAcMain.WaitTimeFragment;

import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
import static com.snowdango.numac.NumAcMain.NumAcActivity.builder;
import static com.snowdango.numac.NumAcMain.NumAcActivity.list;
import static com.snowdango.numac.NumAcMain.NumAcActivity.metrics;
import static com.snowdango.numac.NumAcMain.NumAcActivity.windowManager;

public class AppListViewActivity extends AppCompatActivity {


    @SuppressLint("SourceLockedOrientationActivity")
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(windowManager == null ){
            windowManager = (WindowManager)getSystemService(WINDOW_SERVICE);
            metrics = new DisplayMetrics();
            Display display = windowManager.getDefaultDisplay();
            display.getMetrics(metrics);
        }

        builder = new AlertDialog.Builder(this);
        setContentView(R.layout.activity_app_list);
        setRequestedOrientation(SCREEN_ORIENTATION_PORTRAIT);
    }


    public void onResume() {
        super.onResume();
        loadAppList();
    }

    @Override
    public void onPause(){
        super.onPause();
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(getApplication(), NumAcActivity.class);
        startActivity(intent);
    }

    public void loadAppList(){
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.addToBackStack(null);
        if (list == null) {
            fragmentTransaction.replace(R.id.fragment_app_list, new WaitTimeFragment());
        } else {
            fragmentTransaction.replace(R.id.fragment_app_list, new AppListViewFragment());
        }
        fragmentTransaction.commit();
    }

}

File: app/src/main/java/com/snowdango/numac/AppListView/AppListViewActivity.kt
package com.snowdango.numac.AppListView

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.WindowManager
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.snowdango.numac.NumAcMain.NumAcActivity
import com.snowdango.numac.NumAcMain.WaitTimeFragment
import com.snowdango.numac.R

class AppListViewActivity : AppCompatActivity() {
    @SuppressLint("SourceLockedOrientationActivity")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (NumAcActivity.windowManager == null) {
            NumAcActivity.windowManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
            NumAcActivity.metrics = DisplayMetrics()
            val display = NumAcActivity.windowManager!!.defaultDisplay
            display.getMetrics(NumAcActivity.metrics)
        }
        NumAcActivity.builder = AlertDialog.Builder(this)
        setContentView(R.layout.activity_app_list)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
    }

    public override fun onResume() {
        super.onResume()
        loadAppList()
    }

    public override fun onPause() {
        super.onPause()
    }

    override fun onBackPressed() {
        val intent = Intent(application, NumAcActivity::class.java)
        startActivity(intent)
    }

    private fun loadAppList() {
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.addToBackStack(null)
        if (NumAcActivity.list == null) {
            fragmentTransaction.replace(R.id.fragment_app_list, WaitTimeFragment())
        } else {
            fragmentTransaction.replace(R.id.fragment_app_list, AppListViewFragment())
        }
        fragmentTransaction.commit()
    }
}